engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1321,6 +1361,21 @@ public StreamingContainerAgent getContainerAgent(String containerId)
     return this.containers.values();
   }
 
+  /*
+   * Get event groupId from,
+   * Operator heartbeat if provided by operator, or else redeploy info available at StrAM
+   */
+  private ContainerRedeployRequest getStopOperatorGroupId(PTOperator oper, OperatorHeartbeat ohb)
+  {
+    for (ContainerRedeployRequest request : redeployRequests.values()) {
+      if (request.getOpeartorsToUndeploy().contains(oper.getId())) {
+        request.getOpeartorsToUndeploy().remove(oper.getId());
```

Comment:
This function is called when operator status is pending_undeploy.  Over there, you are also adding this operator_id to deploySet. Hence, It will be more readable to remove this operator from undeploySet at that location rather than in this function. 
Also, the function name needs change
The function doesnt use OperatorHeartBeat ohb object.